### ISSUE TITLE
Types not correct for adding custom levels to configure object

### ISSUE THREAD

**Issue by kevintechie** [log4js-node/log4js-node#1347]
When configuring custom log levels in TypeScript:

```
log4js.configure({
  levels: {
    VERBOSE: { value: 10500, colour: 'cyan' },
    SILLY: { value: 4500, colour: 'blue' },
  },
  appenders: {
    console: { type: 'console', layout },
  },
  categories: {
    testApp: { appenders: ['console'], level: 'info' },
    default: { appenders: ['console'], level: 'info' },
  },
});

const log = log4js.getLogger('testApp');
log.level = 'silly';

log.debug('test 123');
log.verbose('asdf');
log.silly('asdfasdfasdf');
```

I get the following type errors:

On the configuration object:
```
No overload matches this call.
  Overload 1 of 2, '(filename: string): Log4js', gave the following error.
    Argument of type '{ levels: { VERBOSE: { value: number; colour: string; }; SILLY: { value: number; colour: string; }; }; appenders: { console: { type: string; }; }; categories: { testApp: { appenders: string[]; level: string; }; default: { ...; }; }; }' is not assignable to parameter of type 'string'.
  Overload 2 of 2, '(config: Configuration): Log4js', gave the following error.
    Type '{ VERBOSE: { value: number; colour: string; }; SILLY: { value: number; colour: string; }; }' is not assignable to type 'Levels'.
      Object literal may only specify known properties, and 'VERBOSE' does not exist in type 'Levels'.ts(2769)
log4js.d.ts(409, 3): The expected type comes from property 'levels' which is declared here on type 'Configuration'
```

On the Logger object:

```
Property 'verbose' does not exist on type 'Logger'.ts(2339)
Property 'silly' does not exist on type 'Logger'.ts(2339)
```

The code actually works, but the type errors are flagged by the IDE -- vscode in my case.

**Comment by kevintechie** [comment]
It should be possible to have dynamic methods that are type safe. However, it may be non-trivial to implement and may change how custom levels are created. The interesting thing is that it currently "works" if you ignore the error, so there should be a way to declare the type in such a way to prevent the type error. Hopefully avoiding "any".

**Comment by lamweili** [comment]
@kevintechie, can you check if the custom levels' type issue is resolved?
- `Configuration` (#1348) - to be released in `6.7.1`
- `Logger` (https://github.com/log4js-node/log4js-node/issues/1347#issuecomment-1316722109)

**Comment by lamweili** [comment]
@kevintechie, I have updated https://github.com/log4js-node/log4js-node/issues/1347#issuecomment-1316722109.
Can you check if the custom levels' type issue is resolved?

**Comment by kevintechie** [comment]
@lamweili, The custom level type issue has been resolved. Thank you.

**Comment by lamweili** [comment]
Released in `log4js@6.7.1`.
